src/Surfnet/StepupSelfService/SamlStepupProviderBundle/Provider/ViewConfig.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SamlStepupProviderBundle\Provider;
+
+use Surfnet\StepupSelfService\SelfServiceBundle\Exception\LogicException;
+
+class ViewConfig
+{
+    /**
+     * @var string
+     */
+    private $loa;
+
+    /**
+     * @var string
+     */
+    private $logo;
+
+    /**
+     * @var array
+     */
+    private $alt;
+
+    /**
+     * @var array
+     */
+    private $title;
+
+    /**
+     * @var array
+     */
+    private $description;
+
+    /**
+     * @var array
+     */
+    private $buttonUse;
+
+    /**
+     * @var array
+     */
+    private $initiateTitle;
+
+    /**
+     * @var array
+     */
+    private $initiateButton;
+
+    /**
+     * @var array
+     */
+    private $explanation;
+
+    /**
+     * @var array
+     */
+    private $authnFailed;
+
+    /**
+     * @var array
+     */
+    private $popFailed;
+
+    /**
+     * @var null
+     */
+    public $currentLanguage = null;
```

Comment:
It's better to solve this problem in the interface instead of having to modify state from outside this object.

Can't we solve this on the DI level? For example, in SurfnetStepupSelfServiceSamlStepupProviderExtension:

```
$viewConfigDefinition = new Definition('Surfnet\StepupSelfService\SamlStepupProviderBundle\Provider\ViewConfig', [
    new Reference('request_stack'),
    $configuration['view_config']['loa'],
    ...
]);
```